README.md

```diff
@@ -2,302 +2,708 @@
 [![Build Status](https://travis-ci.org/ismasan/parametric.png)](https://travis-ci.org/ismasan/parametric)
 [![Gem Version](https://badge.fury.io/rb/parametric.png)](http://badge.fury.io/rb/parametric)
 
-DSL for declaring allowed parameters with options, regexp pattern and default values.
+Utilities for declaring and manipulating attribute schemas, including custom validations and coercions.
 
-Useful for building self-documeting APIs, search or form objects.
+Useful for building self-documeting APIs, search or form objects. Or possibly as an alternative to Rails' _strong parameters_ (it has no dependencies on Rails and can be used stand-alone).
 
-## Usage
+## Schema
 
-Declare your parameters
+Define a schema
 
 ```ruby
-class OrdersSearch
-  include Parametric::Params
-  param :q, 'Full text search query'
-  param :page, 'Page number', default: 1
-  param :per_page, 'Items per page', default: 30
-  param :status, 'Order status', options: ['checkout', 'pending', 'closed', 'shipped'], multiple: true
+schema = Parametric::Schema.new do
+  field(:title).type(:string).present
```

Comment:
I can't help thinking that the `.type` policy is a bit verbose. It ends up getting used on every field. Would it be "good" to split the single type policy out into one policy per type? (eg. instead of `.type(:string)` just `.string`?) I'm not sure what the consequences might be